jointcal/convert_ci_hsc_jointcal_to_exposurecatalogs.py

```diff
@@ -0,0 +1,69 @@
+import lsst.afw.table
+import lsst.afw.image
+import lsst.afw.geom
+import glob
+import os
+import re
+
+
+def convert():
+    jointcalDir = os.path.join(os.environ['TESTDATA_CI_HSC_DIR'], 'jointcal')
+
+    photoCalibFiles = sorted(glob.glob(os.path.join(jointcalDir, 'jointcal_photoCalib-*.fits')))
+    wcsFiles = sorted(glob.glob(os.path.join(jointcalDir, 'jointcal_wcs-*.fits')))
+
+    photoCalibMap = {}
+    for photoCalibFile in photoCalibFiles:
+        matches = re.search(r'photoCalib-(\d+)-(\d+)\.fits', photoCalibFile)
+        photoCalibMap[photoCalibFile] = (int(matches.groups()[0]),
+                                         int(matches.groups()[1]))
+
+    wcsMap = {}
+    for wcsFile in wcsFiles:
+        matches = re.search(r'wcs-(\d+)-(\d+)\.fits', wcsFile)
+        wcsMap[wcsFile] = (int(matches.groups()[0]),
+                           int(matches.groups()[1]))
+
+    expCatalogSchema = lsst.afw.table.ExposureTable.makeMinimalSchema()
+    expCatalogSchema.addField('visit', type='I', doc='Visit number')
+    expCatalogSchema.addField('detector_id', type='I', doc='Detector number')
+
+    expCatalog = None
+    lastVisit = None
+    for photoCalibFile in photoCalibFiles:
+        if photoCalibMap[photoCalibFile][0] != lastVisit:
+            if expCatalog is not None:
+                expCatalog.writeFits(f'jointcal_photoCalibCatalog-{lastVisit:07d}.fits')
+
+            # Make a new expCatalog
+            expCatalog = lsst.afw.table.ExposureCatalog(expCatalogSchema)
+            lastVisit = photoCalibMap[photoCalibFile][0]
+
+        rec = expCatalog.addNew()
+        rec['visit'] = photoCalibMap[photoCalibFile][0]
+        rec['detector_id'] = photoCalibMap[photoCalibFile][1]
+
+        photoCalib = lsst.afw.image.PhotoCalib.readFits(photoCalibFile)
+        rec.setPhotoCalib(photoCalib)
```

Comment:
I don't understand what is being done here, but that might just be from me not working with these objects before. It looks like each visit has an `ExposureCatalog`, which contains a record for each detector ID. Since the files are sorted, they should be grouped by visit so that all of the files with the same visit will be read consecutively. So you only write the `ExposureCatalog` for each visit once you've moved to a filename for a new visit, which includes a record for all of the detector IDs for that visit. But if that's correct, when does the last `ExposureCatalog` get written? Am I missing something?

The same for the next block for the wcsFiles.

Reply:
Good catch!  Now I'm wondering how the tests passed... or it might be that the tests aren't comprehensive enough to notice something missing.  I will investigate (and fix this).